### ISSUE TITLE
Left tab description

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#278]
I have noticed on enwiki the following for the top left tab
![tab1](https://user-images.githubusercontent.com/861631/48418980-ae262180-e74e-11e8-9985-1df3df06a61c.jpg)
![tab2](https://user-images.githubusercontent.com/861631/48418990-b2ead580-e74e-11e8-8081-702ac29c0b96.jpg)
![tab3](https://user-images.githubusercontent.com/861631/48419001-b7af8980-e74e-11e8-849f-f09ae601e88c.jpg)
![tab4](https://user-images.githubusercontent.com/861631/48419116-065d2380-e74f-11e8-9070-b4ed2f09f012.jpg)

Whereas xowa shows all pages regardless of namespace like the last image (shows Article)
My new branch div-ns is an attempt to reproduce the above behaviour in xowa
(I believe that the changes will work for other language wikis as well)

P.S.
Quite sneeky putting some of the texts in the Mediawiki namespace

**Comment by gnosygnu** [comment]
Cool. I looked at https://github.com/desb42/xowa/commit/d0a8e7f83e3f8c0fbc1fa5f22b574d66f7402ccc and that makes sense.

I'll add it to my todo list (Sorry: I know I'm pretty far behind in merging your commits)

**Comment by desb42** [comment]
Who needs sleep? (I am amazed at the current level of responsiveness)

**Comment by gnosygnu** [comment]
Yeah, I could use more sleep. :)

Also, honestly, you (and everyone else) should also get more responses from me. I'd really like to get through the various comments and issues, but I've had a lot going on in my life recently.

Looking to try to catch up during late evenings and weekends. In the meantime, do bump issues or let me know if I missed anything.

Thanks.

**Comment by desb42** [comment]
On looking at my copy of commons.wikimedia.org/wiki/Main_Page, I get
![commons_tab](https://user-images.githubusercontent.com/861631/48610328-9b912f80-e97c-11e8-9304-7f35e9a85cde.jpg)

Note a blank left tab

This is because commons does not seem to have a mainpage-nstab Mediawiki entry (In fact I dont know how the real site determines this tab name)

Another confusion is that there are entries in the Mediawiki namespace that infer that what I thought were text messages (that is, constant/unchanging) need to be parsed. For example
commons.wikimedia.org/wiki/Mediawiki:nstab-main/de contains

    {{#ifeq:{{FULLPAGENAME}}|Hauptseite|Hauptseite|Galerie}}

Although I do not want to abandon these changes, more thought is needed

**Comment by desb42** [comment]
On further review I see that in the english language gfs (eg en.gfs) is a line
`mainpage-nstab|`
setting the default to empty
If I look at the equivalent mediawiki en.json it is:
`"mainpage-nstab": "Main page",`
When I set en.gfs to have `Main page` then I get the results I was hoping for

I suspect that the en.gfs was 'generated' from en.json at some time in the past. Maybe it needs another review?

**Comment by gnosygnu** [comment]
Yeah, I haven't refreshed the descriptions in over a year. Your workaround above is okay in the meantime. Thanks.

**Comment by gnosygnu** [comment]
> Apart from the fact that they are in English (see #278), they are in the wrong order

BTW: Sorry for missing that. Added that issue to `To investigate` lane, and will take a look at later

Feel free to bump / mention any other missed issues in past.
